spring-web/src/main/java/org/springframework/http/server/reactive/JettyCoreServerHttpResponse.java

```diff
@@ -0,0 +1,315 @@
+import java.util.Map;
+
+import org.eclipse.jetty.http.HttpCookie;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.io.ByteBufferPool;
+import org.eclipse.jetty.io.Content;
+import org.eclipse.jetty.io.RetainableByteBuffer;
+import org.eclipse.jetty.server.HttpCookieUtils;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.IteratingCallback;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.ZeroCopyHttpOutputMessage;
+import org.springframework.http.support.JettyHeadersAdapter;
+import org.springframework.lang.Nullable;
+
+/**
+ * Adapt an Eclipse Jetty {@link Response} to a {@link org.springframework.http.server.ServerHttpResponse}.
+ *
+ * @author Greg Wilkins
+ * @author Lachlan Roberts
+ * @since 6.2
+ */
+class JettyCoreServerHttpResponse extends AbstractServerHttpResponse implements ZeroCopyHttpOutputMessage {
+
+	private final Response response;
+
+	public JettyCoreServerHttpResponse(Response response) {
+		this(null, response);
+	}
+
+	public JettyCoreServerHttpResponse(@Nullable DefaultDataBufferFactory dataBufferFactory, Response response) {
+		super(dataBufferFactory == null ? DefaultDataBufferFactory.sharedInstance : dataBufferFactory,
+				new HttpHeaders(new JettyHeadersAdapter(response.getHeaders())));
+		this.response = response;
+
+		// remove all existing cookies from the response and add them to the cookie map, to be added back later
+		for (ListIterator<HttpField> i = this.response.getHeaders().listIterator(); i.hasNext(); ) {
+			HttpField f = i.next();
+			if (f instanceof HttpCookieUtils.SetCookieHttpField setCookieHttpField) {
+				HttpCookie httpCookie = setCookieHttpField.getHttpCookie();
+				ResponseCookie responseCookie = ResponseCookie.from(httpCookie.getName(), httpCookie.getValue())
+						.httpOnly(httpCookie.isHttpOnly())
+						.domain(httpCookie.getDomain())
+						.maxAge(httpCookie.getMaxAge())
+						.sameSite(httpCookie.getSameSite().name())
+						.secure(httpCookie.isSecure())
+						.build();
+				this.addCookie(responseCookie);
+				i.remove();
+			}
+		}
+	}
+
+	@Override
+	protected Mono<Void> writeWithInternal(Publisher<? extends DataBuffer> body) {
+		return Flux.from(body)
+				.flatMap(this::sendDataBuffer, 1)
+				.then();
+	}
+
+	@Override
+	protected Mono<Void> writeAndFlushWithInternal(Publisher<? extends Publisher<? extends DataBuffer>> body) {
+		return Flux.from(body).flatMap(this::writeWithInternal, 1).then();
```

Comment:
Jetty writes at this level are non-buffering.  Any aggregation that might require flushing is done at a higher level.     Now, we might actually buffer (e.g. if sitting in a flow control congested HTTP/2 stream), but if so, then flushing is not going to help.

The only time we really need a flush semantic is if there is no content at all, then we will write an empty buffer, so that at least the HTTP response headers are generated etc. So perhaps this method needs to detect if the body was actually empty, in which case an empty buffer should be written